src/test/java/com/cloudbees/plugins/credentials/CredentialsProviderTypeRestrictionTest.java

```diff
@@ -0,0 +1,43 @@
+package com.cloudbees.plugins.credentials;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.recipes.LocalData;
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public class CredentialsProviderTypeRestrictionTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    @LocalData
```

Comment:
I don't understand what's going on here,

During the test:

![image](https://user-images.githubusercontent.com/21194782/135808774-ce3bf337-c389-41ba-9aeb-d15702183b6d.png)

At runtime:

![image](https://user-images.githubusercontent.com/21194782/135809280-67083082-c55f-4b73-bba3-a57c9dad2192.png)

Last night when I was playing with it at runtime it was working fine, the descriptor was loaded properly and the adapting code was working just fine =/

Reply:
After wiping my work directory it's back to working at runtime:

![image](https://user-images.githubusercontent.com/21194782/135812057-55d4ec9b-e60d-4479-a3dc-73af35419889.png)

Reply:
Actually I'll just use the dummy credentials as that seems to be what other tests do, still weird it doesn't work